src/TableV8/index.tsx

```diff
@@ -98,23 +99,23 @@ export const TableV8 = <TData extends object>({
   withStripes = false,
   withTabs = false,
 }: ReactTableProps<TData>) => {
-  const table = useReactTable({
+  const reactTable = useReactTable({
     columnResizeMode: 'onChange',
     columns,
     data,
     enableColumnResizing,
     enableSorting,
     getCoreRowModel: getCoreRowModel(),
-    initialState: { pagination: { pageSize: DEFAULT_TABLE_PAGE_SIZE } },
-    ...(manualPagination
+    initialState: { pagination: { pageSize: DEFAULT_TABLE_PAGE_SIZE }, ...initialState },
+    ...(withPagination && manualPagination
       ? { manualPagination, onPaginationChange, pageCount }
       : { getPaginationRowModel: getPaginationRowModel() }),
-    ...(manualSorting
+    ...(enableSorting && manualSorting
       ? { manualSorting, onSortingChange }
       : { getSortedRowModel: getSortedRowModel() }),
     state: {
-      ...(manualPagination ? { pagination: paginationState } : {}),
-      ...(manualSorting ? { sorting: sortingState } : {}),
+      ...(withPagination && manualPagination ? { pagination: paginationState } : {}),
+      ...(enableSorting && manualSorting ? { sorting: sortingState } : {}),
       ...(withPagination ? {} : singlePagePaginationState),
```

Comment:
I think all of these ternaries can just use `&&`. They all appear to be boolean flags for the table and all the false conditions are `{}` empty objects. So spreading empty objects just returns undefined so you'll get nothing because it's using Object.assign.

Reply:
withPagination && manualPagination && { pagination: paginationState } gives me an error because I can't spread a non-iterable (false) value.

if pagination isn't manually controlled then I don't want to provide pagination state - react table will take care of it